hummingbot/connector/exchange/btc_markets/btc_markets_exchange.py

```diff
@@ -0,0 +1,966 @@
+        socket API updates.
+        """
+        while True:
+            try:
+                self._poll_notifier = asyncio.Event()
+                await self._poll_notifier.wait()
+
+                await safe_gather(
+                    self._update_balances(),
+                    self._update_order_status(),
+                )
+                self._last_poll_timestamp = self.current_timestamp
+            except asyncio.CancelledError:
+                raise
+            except Exception as e:
+                self.logger().error(str(e), exc_info=True)
+                self.logger().network("Unexpected error while fetching account updates.",
+                                      exc_info=True,
+                                      app_warning_msg="Could not fetch account updates from BTCMarkets.net. "
+                                                      "Check API key and network connection.")
+                await asyncio.sleep(0.5)
+
+    async def _update_balances(self):
+        """
+        Calls REST API to update total and available balances.
+        """
+        local_asset_names = set(self._account_balances.keys())
+        remote_asset_names = set()
+        account_info = await self._api_request("GET", f"{Constants.ACCOUNTS_URL}/me/balances", {}, True)
+        for account in account_info:
+            asset_name = account["assetName"]
+            self._account_available_balances[asset_name] = Decimal(str(account["available"]))
+            self._account_balances[asset_name] = Decimal(str(account["balance"]))
+            # self._account_locked_balances[asset_name] = Decimal(str(account["locked"]))
+            remote_asset_names.add(asset_name)
+
+        asset_names_to_remove = local_asset_names.difference(remote_asset_names)
+        for asset_name in asset_names_to_remove:
+            del self._account_available_balances[asset_name]
+            del self._account_balances[asset_name]
+
+        # as self._real_time_balance_update = false
+        # take `in_flight_orders` snapshot during `_update_balances` (see #2224)
+        self._in_flight_orders_snapshot = {k: copy.copy(v) for k, v in self._in_flight_orders.items()}
+        self._in_flight_orders_snapshot_timestamp = btc_markets_utils.get_ms_timestamp()
+
+    async def _update_order_status(self):
+        """
+        Calls REST API to get status update for each in-flight order.
+        """
+
+        last_tick = int(self._last_poll_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL)
+        current_tick = int(self.current_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL)
+
+        if current_tick > last_tick and len(self._in_flight_orders) > 0:
+            tracked_orders = list(self._in_flight_orders.values())
+            tasks = []
+            for tracked_order in tracked_orders:
+                order_id = await tracked_order.get_exchange_order_id()
```

Comment:
The method `get_exchange_order_id()` can raise `asyncio.TimeoutError` and it should be handled

Reply:
based on crypto.com connector - https://github.com/hummingbot/hummingbot/blob/master/hummingbot/connector/exchange/crypto_com/crypto_com_exchange.py#L602